src/PlusDataCollection/Andor/vtkPlusAndorCamera.h

```diff
@@ -0,0 +1,266 @@
+{
+public:
+  /*! Constructor for a smart pointer of this class*/
+  static vtkPlusAndorCamera* New();
+  vtkTypeMacro(vtkPlusAndorCamera, vtkPlusDevice);
+  virtual void PrintSelf(ostream& os, vtkIndent indent) VTK_OVERRIDE;
+
+  /*! Specify the device connected to this class */
+  virtual bool IsTracker() const
+  {
+    return false;
+  }
+
+  /*! Read configuration from xml data */
+  virtual PlusStatus ReadConfiguration(vtkXMLDataElement* config);
+
+  /*! Write configuration to xml data */
+  virtual PlusStatus WriteConfiguration(vtkXMLDataElement* config);
+
+  /*! Verify the device is correctly configured */
+  virtual PlusStatus NotifyConfigured();
+
+  /*! Get the version of SDK */
+  virtual std::string GetSdkVersion();
+
+  /*! Shutter mode:
+   * 0 Fully Auto
+   * 1 Permanently Open
+   * 2 Permanently Closed
+   * 4 Open for FVB series
+   * 5 Open for any series
+   *
+   * For an external shutter: Output TTL high signal to open shutter.
+   */
+  PlusStatus SetShutter(int shutter);
+  int GetShutter();
+
+  /*! Frame exposure time, seconds. Sets to the nearest valid value not less than the given value. */
+  PlusStatus SetExposureTime(float exposureTime);
+  float GetExposureTime();
+
+  /*! Horizontal and vertical binning. Allowed values: 1, 2, 4, 8. */
+  PlusStatus SetHorizontalBins(int bins);
+  PlusStatus SetVerticalBins(int bins);
+
+  /*! Horizontal and vertical shift speed. */
+  PlusStatus SetHSSpeed(int type, int index);
+  PlusStatus SetVSSpeed(int index);
+
+  /*! Index of the pre-amp gain, not the actual value. */
+  PlusStatus SetPreAmpGain(int preAmptGain);
+  int GetPreAmpGain();
+
+  /*! Acquisition mode. Valid values:
+   * 1 Single Scan
+   * 2 Accumulate
+   * 3 Kinetics
+   * 4 Fast Kinetics
+   * 5 Run till abort
+   */
+  PlusStatus SetAcquisitionMode(int acquisitionMode);
+  int GetAcquisitionMode();
+
+  /*! Readout mode. Valid values:
+   * 0 Full Vertical Binning
+   * 1 Multi-Track
+   * 2 Random-Track
+   * 3 Single-Track
+   * 4 Image
+   */
+  PlusStatus SetReadMode(int setReadMode);
+  int GetReadMode();
+
+  /*! Trigger mode. Valid values:
+   * 0. Internal
+   * 1. External
+   * 6. External Start
+   * 7. External Exposure (Bulb)
+   * 9. External FVB EM (only valid for EM Newton models in FVB mode)
+   * 10. Software Trigger
+   * 12. External Charge Shifting
+   */
+  PlusStatus SetTriggerMode(int triggerMode);
+  int GetTriggerMode();
+
+  /*! Normal operating temperature (degrees celsius). */
+  PlusStatus SetCoolTemperature(int coolTemp);
+  int GetCoolTemperature();
+
+  /*! Lowest temperature at which it is safe to shut down the camera. */
+  PlusStatus SetSafeTemperature(int safeTemp);
+  int GetSafeTemperature();
+
+  /*! Get the current temperature of the camera in degrees celsius. */
+  float GetCurrentTemperature();
+
+  /*! Paths to additive and multiplicative bias correction images. */
+  PlusStatus SetBiasCorrectionImage(std::string biasFilePath);

```

Comment:
Even though the compiler would probably optimize it, should this be const &?

Reply:
It should be `const`, but perhaps not `const &`.